### ISSUE TITLE
Support printing to markdown

### ISSUE THREAD

**Issue by shonfeder** [ocaml/omd#231]
This would allow programmatic construction and transformation of markdown, and (in the special case of a no-op transformation) would serve as a markdown pretty printer.

Request by @jfrolich on at https://github.com/ocaml/omd/issues/230#issuecomment-799033420

**Comment by cuihtlauac** [comment]
Not sure if it is already known. ocaml-lsp-server is stuck with 1.3.2:
```
(omd (and (>= 1.3.2) (< 2.0.0~alpha1)))
```
It's using `Omd.to_markdown` in doc_to_md.ml

I believe this is problematic.

**Comment by shonfeder** [comment]
@cuihtlauac Could you explain what kind of problems you're hitting on 1.3.2?

**Comment by cuihtlauac** [comment]
Because `ocaml-lsp-server` requires `omd < 2` it is not possible to have it installed in the same switch as packages requiring `omd >= 2`. Being forced to choose between mutually exclusive dependencies is a problem, I believe.

**Comment by shonfeder** [comment]
Thanks for explaining. I think this issue is certainly top priority at the moment.

My suggestion for the moment would be to avoid depending on the pre-release versions of 2.0 for any production-grade projects. As noted in the README, we have not yet achieved feature parity with 1.3, and only just recently resolved the lingering bugs causing failures to satisfy the CommonMark spec.

**Comment by cuihtlauac** [comment]
You're right. Unfortunately, a number of projects have decided not to wait and are relying on 2.0 features, they probably had good reasons too.
